spark/src/test/java/org/apache/iceberg/spark/source/TestParquetWrite.java

```diff
@@ -339,7 +342,8 @@ public void testPartitionedCreateWithTargetFileSizeViaOption() throws IOExceptio
     Assert.assertTrue("All DataFiles contain 1000 rows", files.stream().allMatch(d -> d.recordCount() == 1000));
   }
 
-  @Test
+  // This fails due to SPARK-28730
```

Comment:
Why do these tests fail? Can you be more specific?

Reply:
The problem here is that Spark's query analysis requires that all columns are present. Getting this to work will require a change to Spark to allow omitting optional columns, and a table capability to turn it on.